src/BatteryStateOfChargeService.cpp

```diff
@@ -43,21 +48,48 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalCurrent_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (newCurrent_ <= 0)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    return estimatedTime_;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
-{
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+{   
+
+    QTime newTime = batteryData.time;
+    QTime oldTime = newTime;
+    double new_Time = QTime(0, 0, 0).secsTo(newTime);
+    double old_Time = QTime(0, 0, 0).secsTo(oldTime);
```

Comment:
Instead of being ambiguous between `new_Time` and `newTime` it would be better to actually name the difference between the two. How about `newTimeSecs` instead?